### DISCUSSION TITLE
tabs sorting

### DISCUSSION THREAD

**Discussion by bsw0314** [sumatrapdfreader/sumatrapdf#4515] [Ideas]
I wish Sumatrapdf could support tabs sorting, it would be very helpful when you are doing literature review. many thanks!
I have searched for similar discussion. Several years ago, someone proposed similar idea. I just wish to make it on table again. This feature is really very helpful.

**Comment by GitHubRulesOK** [comment]
Sorting is not simple as there are different wishes and not all custom sorts would be practical.
 Basically is currently in last viewed order unless dragged.
![image](https://github.com/user-attachments/assets/6f1fbdb8-3633-47fb-9ac4-82e9e59c5a1a)

 So is your desire alphabetical or numerical ?

However tabs may be easy to write into order in different ways, thus depends what is your need.
I have shown that you can extract all current tabs filenames to a file where you could add any external sorting then reload the tabbed list.

Let us assume all those tabs are in this current order how should they be sorted 
By drive ? by folder ? by initial letter of filename ? or simply kept in the current user defined order.
```
SessionData [
[
TabStates [
[
 FilePath = C:\Users\K\Downloads\SU\Super User 5
 FilePath = C:\Users\K\Downloads\SU\SU 1852990 c
 FilePath = C:\Users\K\Downloads\SU\Super User 5
 FilePath = C:\Users\K\Downloads\SU\Super User 5
 FilePath = C:\Users\K\Downloads\SU\Super User 2
 FilePath = C:\Users\K\Downloads\SU\SU 1804137 s
 FilePath = C:\Users\K\Downloads\SU\SU 517986 wi
 FilePath = C:\Users\K\Downloads\SU\Super User 5
 FilePath = C:\Users\K\Downloads\SU\SU 1852990 c
 FilePath = C:\Users\K\Downloads\SU\Super User 2
 FilePath = C:\Users\K\Downloads\SU\Super User 5
 FilePath = C:\Users\K\Downloads\SU\SU 1737542 w
```

**Reply by bsw0314** [reply]
Hi, thanks for your reply. Yes, I would like to sort tabs alphabetically and/or numerically, just like in windows explorer. By Ctrl+K, and then @, I could see the open tabs list. If this list could be alphabetically and/or numerically sorted, that would be very helpful.

**Reply by GitHubRulesOK** [reply]
There are currently two areas where filenames are sorted one is "Favourites" and the other is `sort pinned documents alphabetically` There is a way to command line script "save all current open tabs" into a file for backing up a session and have shown that several times for different requests. It is also very easy to command line sort a file on a given position such as 3rd character etc but in this case the default of alpha sort should be enough. And until such time as a feature be added to SumatraPDF internally should work well.
To raise a suggestion use the Canny system for others to upvote as well.

```
Desktop>sort /?
SORT [/R] [/+n] [/M kilobytes] [/L locale] [/REC recordbytes]
  [[drive1:][path1]filename1] [/T [drive2:][path2]]
  [/O [drive3:][path3]filename3]
  /+n                         Specifies the character number, n, to
                              begin each comparison.  /+3 indicates that
                              each comparison should begin at the 3rd
                              character in each line.  Lines with fewer
                              than n characters collate before other lines.
                              By default comparisons start at the first
                              character in each line.
```

thus it would be easy to add a single "sort" command to the windows batch file.


For the suggested way to save tabs as they are see https://github.com/sumatrapdfreader/sumatrapdf/discussions/3045#discussioncomment-3941188
